src/services/contract-service.ts

```diff
@@ -0,0 +1,91 @@
+import { Contract } from '@ethersproject/contracts';
+import { Erc20 } from 'models/generated/erc20/Erc20';
+import { Erc721 } from 'models/generated/erc721';
+import { ITokenInfo } from 'services/token-types';
+import { Provider } from '@ethersproject/providers';
+
+import { ContractInterface, Signer, getDefaultProvider } from 'ethers';
+
+import { DI, IContainer, Registration } from 'aurelia';
+import { ICacheService } from './cache-service';
+import { ITokenListUniswap } from './token-types';
+import { cache } from 'decorators/cache';
+import governanceAlfajores from '../contracts/governance/alfajores.json';
+import governanceCelo from '../contracts/governance/celo.json';
+import governanceShared from '../contracts/governance/sharedAbis.json';
+import monetaryAlfajores from '../contracts/monetary/alfajores.json';
+import monetaryCelo from '../contracts/monetary/celo.json';
+import monetaryShared from '../contracts/monetary/sharedAbis.json';
+
+type MonetaryContractJson = typeof monetaryCelo | typeof monetaryAlfajores;
+type GovernanceContractJson = typeof governanceCelo | typeof governanceAlfajores;
+type Shared = typeof governanceShared | typeof monetaryShared;
+type Contracts = GovernanceContractJson['contracts'] | MonetaryContractJson['contracts'];
+export type MonetaryContracts = Extract<keyof MonetaryContractJson['contracts'], string>;
+type GovernanceContracts = Extract<keyof GovernanceContractJson['contracts'], string>;
+
+const endpoint = import.meta.env.KOL_NETWORK === 'Celo' ? 'https://celo.rpcs.dev:8545' : `https://alfajores.rpcs.dev:8545`;
+export const defaultProvider = getDefaultProvider(endpoint);
+
+export type IContractService = ContractService;
+export const IContractService = DI.createInterface<IContractService>('ContractService');
+
+const tokenListUri = 'https://cdn.jsdelivr.net/gh/Kolektivo/tokenlists@main/tokenlist.json';
+
+export const tokenInfos =
```

Comment:
you have removed code from token-service that we are going to need, namely:

 `isRequiredTokenContract` (I have been informed, but we can confirm because I don't recall ATM exactly where in the UI)
 
 `_getTokenInfoFromAddress` which initializes a tokenInfo when it is not found in the tokenlist using `getTokenInfoOnChain` (same need to confirm), and
 
 `getTokenAddressId` that is involved in caching the tokenInfos.

Reply:
I am not real wild about shoving so many domains of code into a single "contract-service".

Reply:
split and shrunk